[docs] clarify that macros create new bindings (#3284)
jump-dev/JuMP.jl: docs/src/manual/variables.md

## Create a variable

Create variables using the [`@variable`](@ref) macro:

```jldoctest variables
julia> model = Model();

> julia> @variable(model, x)

Author: ```suggestion
julia> @variable(model, x)
x
```

> julia> typeof(x)

Author: ```suggestion
julia> typeof(x)
VariableRef
```

## Create a variable

Create variables using the [`@variable`](@ref) macro:

```jldoctest variables
julia> model = Model();

julia> @variable(model, x)
x

julia> typeof(x)
VariableRef

julia> num_variables(model)
1
```

> Here `x` is a Julia variable that is bound to the JuMP `VariableRef` struct, and

Author: ```suggestion
Here `x` is a Julia variable that is bound to a JuMP `VariableRef` object, and
```
